lib/orm/associations/belongs-to.js

```diff
@@ -13,7 +15,13 @@ import assert from "../../assert";
  */
 export default class BelongsTo extends Association {
   get identifier() {
-    return `${camelize(this.key)}Id`;
+    if (typeof identifierCache[this.key] === "string")
+      return identifierCache[this.key];
+
+    const identifier = `${camelize(this.key)}Id`;
+
+    identifierCache[this.key] = identifier;
+    return identifier;
   }
```

Comment:
I'd like to avoid early and double returns in these methods if possible. I'm thinking something like this might be a bit more readable.

```js
get identifier() {
  if (typeof identifierCache[this.key] !== "string") {
    const identifier = `${camelize(
      this._container.inflector.singularize(this.key)
    )}Ids`;

    identifierCache[this.key] = identifier;
  }

  return identifierCache[this.key];
}
```

Is there any downside to setting the cache if the value doesn't exist, and then returning from the cache?

Reply:
I don't think that there's a downside. 
I just try to avoid the additional nesting (which is probably cleaner in other languages, see e.g. [kotlin](https://github.com/Turbo87/intellij-emberjs/blob/27d45ad1b50f1f1476b9cd50f07abac77e32bca2/src/main/kotlin/com/emberjs/psi/EmberRelationshipAnnotator.kt#L29))

If you usually do it that way, I'll refactor it to match your style.

Reply:
Yup we usually write in this style so I think it makes sense to refactor these.

Reply:
@ryanto I've commited some changes. 
Can you check if I understood your request correctly? I'll squash it after it's ok.

Reply:
Yup, looks great 👍